src/test/java/org/broadinstitute/consent/http/service/dao/DatasetServiceDAOTest.java

```diff
@@ -725,6 +725,30 @@ void testPatchDataset() throws Exception {
     assertEquals(3, patched.getProperties().size());
   }
 
+  @Test
+  void testPatchDatasetWithNullName() throws Exception {
+    Dataset dataset = createDataset();
+    User user = userDAO.findUserById(dataset.getCreateUserId());
+    DatasetPatch patch = new DatasetPatch(null, List.of());
+    serviceDAO.patchDataset(dataset.getDatasetId(), user, patch);
+    Dataset patched = datasetDAO.findDatasetById(dataset.getDatasetId());
```

Comment:
Have you thought about making a fully mocked test for this class? It seems like you could mock the DAOs and JDBI so a test doesn't need to DB fetch to verify the code. That could also help narrow the scope of a test.

Reply:
TBH, no, we haven't tried that. I find more bugs by testing actual DB integration than in our other testing layers.